test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -1117,5 +1117,33 @@ void testLoopNestComputeAt_4() {
   // TODO: Verify that computeAt works with reduction axis
 }
 
+void testOuterLoopVectorization() {
+  KernelScope kernel_scope;
+  Tensor* tensor = Compute(
+      "f", {{8, "X"}, {8, "y"}}, [](const VarHandle& x, const VarHandle& y) {
+        return ExprHandle(1.0f) + cast<float>(x) * x + cast<float>(y) * y;
+      });
+  LoopNest l({tensor});
+
+  Stmt* outer_loop = *dynamic_cast<Block*>(l.root_stmt())->stmts().begin();
+  l.vectorize(dynamic_cast<For*>(outer_loop));
+
+  Stmt* root_stmt = l.root_stmt();
+  Block* outer_block = dynamic_cast<Block*>(root_stmt);
+  ASSERT_NE(outer_block, nullptr);
+  while (Block* inner_block = dynamic_cast<Block*>(*outer_block->stmts().begin())) {
+    outer_block = inner_block;
+  }
+
+  ASSERT_EQ(outer_block->nstmts(), 1);
+  For* for_loop = dynamic_cast<For*>(*outer_block->stmts().begin());
+  ASSERT_NE(for_loop, nullptr);
+  Block* for_body = dynamic_cast<Block*>(for_loop->body());
```

Comment:
Return type of `For::body` is `Block*`, so there is no need in this dynamic cast.